preview/preview.go

```diff
@@ -0,0 +1,131 @@
+package preview
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	logger "github.com/ipfs/go-log"
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/dataset/dsio"
+	"github.com/qri-io/qfs"
+)
+
+var (
+	log = logger.Logger("preview")
+)
+
+const (
+	// MaxNumDatasetRowsInPreview is the highest number of rows a dataset preview
+	// can contain
+	MaxNumDatasetRowsInPreview = 100
+	// MaxReadmePreviewBytes determines the maximum amount of bytes a readme
+	// preview can be. three bytes less than 1000 to make room for an elipsis
+	MaxReadmePreviewBytes = 997
+)
+
+// CreatePreview generates a preview for a dataset version
+// It expects the passed in dataset to have any relevant script files already
+// loaded
+// Preview currently includes:
+//    - body: 100 rows
+//    - readme: first 997 bytes
+//    - meta: all
+//    - commit: all
+//    - structure: all
+//    - stats: none
+//    - viz: all
+//    - transform: all
+func CreatePreview(ctx context.Context, ds *dataset.Dataset) (*dataset.Dataset, error) {
+
+	var err error
+
+	if ds == nil || ds.IsEmpty() {
+		log.Debugf("CreatePreview: empty dataset")
+		return nil, fmt.Errorf("empty dataset")
+	}
```

Comment:
is it better to pull these two error cases out and report them separately? 

```
if ds == nil {
    return. nil, fmt.Errorf("nil dataset")
}
if ds.IsEmpty() {
    return nil, fmt.Errorf("empty dataset")
}
```

Reply:
Yeah, might be worth calling them out separately to avoid confusion down the road